pkg/kubeconf/kubelet_configz.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconf
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"time"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	kubeletconfiginternal "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	kubeletscheme "k8s.io/kubernetes/pkg/kubelet/apis/config/scheme"
+)
+
+const (
+	inClusterTokenFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+)
+
+// GetKubeletConfiguration returns the kubelet configuration.
+func GetKubeletConfiguration(restConfig *rest.Config) (*kubeletconfiginternal.KubeletConfiguration, error) {
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(restConfig)
```

Comment:
IIRC in the past we didn't try the discovery client approach, I'd be in favour of trying this way

Reply:
Could you enlighten what the DiscoveryClient is and why we need it?